### PR TITLE
Add U2F dependency

### PR DESCRIPTION
**Pull request by joshraker**
https://github.com/Yubico/libu2f-host#building

Notes:

- The binary can be built for macOS without the `macosx.mk` file.
- Even when building from a release tarball, `gengetopt` and `help2man` are required.

### CODE REVIEW

File: config/software/u2f.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name 'u2f'
+
+unless windows?
+  dependency 'gengetopt'
+  dependency 'help2man'
```
Comment (joshraker) [inline config/software/u2f.rb]: `help2man` is already defined in [omnibus-software](https://github.com/chef/omnibus-software/tree/master/config/software) but `gengetopt` is not.

**Comment by joshraker** [comment]
@UserNotFound @robertfairhead before I go much further, am I handling/adding dependencies correctly? I've not used omnibus before and have to add quite a few dependencies to get U2F to work.

**Comment by UserNotFound** [comment]
I'm not sure, I have zero experience here. This might be a good chance to do something like a lunch-and-learn if we need to build up some internal knowledge to make an effective review here.

**Comment by robertfairhead** [comment]
I've never used omnibus-toolbelt before either, so I don't have enough experience to say offhand.

**Comment by joshraker** [comment]
I'm investigating including `libu2f-host` as a non-embedded dependency. It looks like this might be possible on debian with a [source package control file](https://www.debian.org/doc/debian-policy/ch-controlfields.html).

**Comment by joshraker** [comment]
Should be able to use `homebrew-cask`'s `depends_on formula:` to include the dependency for `libu2f-host` to our `aptible` cask. https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/depends_on.md#depends_on-formula

Unsure if we can include the dependency directly in our `.pkg`.

**Comment by joshraker** [comment]
Adding dependencies to a RH/CentOS package: http://ftp.rpm.org/api/4.4.2.2/dependencies.html

**Comment by joshraker** [comment]
It's unfortunate that it took this long to find that omnibus does support [runtime_dependencies](https://www.rubydoc.info/github/chef/omnibus/Omnibus/Project#runtime_dependency-instance_method) but this does what my script did for the .deb and .rpm packages. Now I just need to figure out how to only add the dependency to supported OS versions. We also need to determine how we want to handle adding the dependency to Debian packages.

Debian: `u2f-host` is not available until 9+ (stretch) but we only build packages for 7 and 8. I was able to install the debian-8 package on 9 without trouble and it can still be installed on 7 or 8 it will just say it has unsatisfied dependencies.

Ubuntu: Available 16.04+ (xenial)

CentOS: Available in 7 but not 8

**Comment by joshraker** [comment]
Found the gem responsible for the platform checks we use (e.g. `ubuntu?`): [chef/sugar](https://github.com/chef/chef-sugar)

**Comment by joshraker** [comment]
I decided to leave the dependency off the Debian packages for now so we don't get a bunch of support requests about the unmet dependency. We should probably add support for newer versions and it will be added to those.

**Pull request merged**